tox.ini

```diff
@@ -32,17 +32,28 @@ exclude = .tox,.cache,.pytest_cache
 [testenv:integrationtests]
 basepython = python3.8
 deps =
-	-r requirements/base.txt
+	-rrequirements/dev.txt
+passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH BRANCH INTEGRATION_TESTS_REPO_URL TARGET_REPO
 setenv =
     API_CLIENT_RUN_ON_REDIS = true
+	INT_TESTS_REPO_SSH_URL = git@github.com:gridsingularity/gsy-backend-integration-tests.git
```

Comment:
I have the feeling that we're creating unnecessary env variables. We have 3 variables that do more or less the same: `TARGET_REPO`, `INTEGRATION_TESTS_REPO_URL` and `INT_TESTS_REPO_URL_SSH`. I understood that the CI server uses HTTPS, while locally we might use SSH. But can't this can be tackled with just one variable?
Suppose we just use `TARGET_REPO`:
- the CI server can fill it with an HTTPS URL (in `development.yaml`, for example);
- for local runs we can fill it with the SSH URL (in `tox.in`).

Do you think this would work?

Reply:
I think is also confusing having so many env variables, specially all nested in order to determine the branch from which is being copied.
However, I'm not sure how that would work. At least it would not be handled automatically and would need to edit tox.ini file while working locally.

I think it could look something like this:
```suggestion
passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH BRANCH INTEGRATION_TESTS_REPO
setenv =
    API_CLIENT_RUN_ON_REDIS = true
    # uncomment for local run using local repository
    # INTEGRATION_TESTS_REPO = /Users/SomeUser/PathToRepo/
allowlist_externals =
	git
	/bin/cp
	/bin/rm
commands_pre =
	pip install -e .
    git clone {env:INTEGRATION_TESTS_REPO:git@github.com:gridsingularity/gsy-backend-integration-tests.git} {envtmpdir}/gsy-backend-integration-tests
```

The thing we should be careful with is to not override the `INTEGRATION_TESTS_REPO` variable set by the workflow with one given inside the testenv, so the tox.ini on the remote repository should always have that one line commented.

I also thought initially for something like this:

```suggestion
passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH BRANCH INTEGRATION_TESTS_REPO
setenv =
    API_CLIENT_RUN_ON_REDIS = true
    # uncomment for local run using local repository
    # INTEGRATION_TESTS_REPO = /Users/SomeUser/PathToRepo/
    # uncomment for local run using remote repository
    # INTEGRATION_TESTS_REPO = git@github.com:gridsingularity/gsy-backend-integration-tests.git
allowlist_externals =
	git
	/bin/cp
	/bin/rm
commands_pre =
	pip install -e .
    git clone {env:INTEGRATION_TESTS_REPO:} {envtmpdir}/gsy-backend-integration-tests
```

The command would look much cleaner, but if uncommenting the appropriate line is forgotten, then it would have nothing set as default.

On the first suggestion, one could deal with uncommenting the line only if interested on running the tests with a local branch.

Even more:

```suggestion
passenv = LANG TERM LANGUAGE LC_ALL LD_LIBRARY_PATH BRANCH INTEGRATION_TESTS_REPO
setenv =
    API_CLIENT_RUN_ON_REDIS = true
allowlist_externals =
	git
	/bin/cp
	/bin/rm
commands_pre =
	pip install -e .
    git clone {env:INTEGRATION_TESTS_REPO:git@github.com:gridsingularity/gsy-backend-integration-tests.git} {envtmpdir}/gsy-backend-integration-tests
```

This should work invoking the testenv as
`INTEGRATION_TESTS_REPO=/Users/SomeUser/PathToRepo/ tox -e integrationtests`
And wouldn't need any local changes of the tox.ini file. 

Which approach do you like the most?
Do you have any further ideas?

Reply:
I believe the last approach is the closest to what you initially described.

Reply:
Addressed [here](https://github.com/gridsingularity/gsy-e-sdk/pull/208#:~:text=of%20environment%20variables.-,442d4f9,-Merge%20state). Will replicate on other repos.

Reply:
But what would then happen if you run `INTEGRATION_TESTS_REPO=/Some/path/ tox -e integrationtests`?
Wouldn't INTEGRATION_TESTS_REPO be overriden inside the testenv with `git@github.com:gridsingularity/gsy-backend-integration-tests.git`, and therefore fail to clone locally?

Reply:
You're right, I was assuming that `setenv` would have been overridden by the value passed by the CLI command, but I was wrong :P I suppose there must be another way, but I couldn't figure it out yet. We can leave it as it is!